addons/stock/models/stock_move.py

```diff
@@ -536,10 +536,14 @@ def _set_lot_ids(self):
                         }))
                         mls_without_lots -= move_line
                     else:  # No line without serial number, creates a new one.
-                        move_line_vals = self._prepare_move_line_vals(quantity=0)
-                        move_line_vals['lot_id'] = lot.id
-                        move_line_vals['lot_name'] = lot.name
-                        move_line_vals['product_uom_id'] = move.product_id.uom_id.id
+                        reserved_quants = self.env['stock.quant']._get_reserve_quantity(move.product_id, move.location_id, 1.0, lot)
+                        if reserved_quants:
+                            move_line_vals = self._prepare_move_line_vals(quantity=0, reserved_quant=reserved_quants[0][0])
+                        else:
+                            move_line_vals = self._prepare_move_line_vals(quantity=0)
+                            move_line_vals['lot_id'] = lot.id
+                            move_line_vals['lot_name'] = lot.name
+                            move_line_vals['product_uom_id'] = move.product_id.uom_id.id
```

Comment:
```suggestion
                        move_line_vals['product_uom_id'] = move.product_id.uom_id.id
```
Might need to keep that one out of the `else`. Otherwise it would set the line in the UoM of the move, which might not be units and cause weird behavior down the line (as we'll have non `1` quantity for serial numbers). Maybe it could be an issue? :thinking: